### DISCUSSION TITLE
Disable deriving `PartialEq`

### DISCUSSION THREAD

**Discussion by fardream** [tokio-rs/prost#599] [Q&A] [answered]
I have a message that will have a customized `impl` for `ParitalEq`. Is there a way to remove `PartialEq` from the `derive` attribute on the generated struct for messages?

For example, I have the following message
```proto
syntax = "proto3";
message SpecialValue {
    int64 value_1 = 1;
    int32 value_2 = 2;
}
```

the generated code will be

```rust
#[derive(Clone, PartialEq, ::prost::Message)]
pub struct SpecialValue {
    #[prost(int64, tag="1")]
    pub value_1: i64,
    #[prost(int32, tag="2")]
    pub value_2: i32,
}
```

I am looking for a way to remove `PartialEq` from the `derive` like below:

```rust
#[derive(Clone, ::prost::Message)]
pub struct SpecialValue {
    #[prost(int64, tag="1")]
    pub value_1: i64,
    #[prost(int32, tag="2")]
    pub value_2: i32,
}
```

**Comment by LucioFranco** [comment, marked as answer]
I don't think there is a way right now but you can always newtype the field and implement your own partialeq which would be easier than including a change in `prost-build` to not derive that attribute.
